datafusion/src/datasource/object_store/mod.rs

```diff
@@ -86,11 +101,17 @@ pub trait ObjectStore: Sync + Send + Debug {
         delimiter: Option<String>,
     ) -> Result<ListEntryStream>;
 
+    /// Get object reader for the path of one file
+    fn file_reader_from_path(&self, file_path: &str) -> Result<Arc<dyn ObjectReader>>;
+
     /// Get object reader for one file
     fn file_reader(&self, file: FileMeta) -> Result<Arc<dyn ObjectReader>>;
 }
 
-static LOCAL_SCHEME: &str = "file";
+lazy_static! {
+    static ref OBJECT_STORES: Box<ObjectStoreRegistry> =
+        Box::new(ObjectStoreRegistry::new());
+}
```

Comment:
In #1072 the consensus seems to be that we should avoid using `static` for the registry.